src/test/kotlin/uk/gov/justice/digital/hmpps/externalusersapi/repository/RoleRepositoryTest.kt

```diff
@@ -49,6 +49,27 @@ class RoleRepositoryTest {
     }
   }
 
+  @Nested
+  inner class FindAutomaticGroupRolesByGroupCode {
+    @Test
+    fun givenAnExistingRoleTheyCanBeRetrieved(): Unit = runBlocking {
+      assertThat(repository.findAutomaticGroupRolesByGroupCode("SITE_1_GROUP_2").toList()).size()
+        .isEqualTo(2)
+    }
+
+    @Test
+    fun givenAnNonAutomaticGroupRoleTheyCantBeRetrieved(): Unit = runBlocking {
+      // SITE_1_GROUP_1 have 2 automatic set to 'true' and 1 set to 'false'
+      assertThat(repository.findAutomaticGroupRolesByGroupCode("SITE_1_GROUP_1").toList()).size()
+        .isEqualTo(2)
+    }
```

Comment:
Do we need a test with a role that doesn't have any assigned groups?

Reply:
added test covering this scenario